src/components/Inngest.ts

```diff
@@ -490,8 +490,22 @@ const builtInMiddleware = (<T extends MiddlewareStack>(m: T): T => m)([
     name: "Inngest: Logger",
     init({ client }) {
       return {
-        onFunctionRun() {
-          const logger = new ProxyLogger(client["logger"]);
+        onFunctionRun(arg) {
+          const { ctx } = arg;
+          const metadata = {
+            runID: ctx.runId,
+            eventName: ctx.event.name,
+            functionName: arg.fn.name,
+          };
+
+          // create a child logger if the provided logger has child logger implementation
+          /* eslint-disable @typescript-eslint/no-unsafe-argument, @typescript-eslint/no-unsafe-assignment */
+          const providedLogger: Logger =
+            "child" in client["logger"]
+              ? client["logger"].child(metadata)
+              : client["logger"];
+          /* eslint-enable @typescript-eslint/no-unsafe-argument, @typescript-eslint/no-unsafe-assignment */
```

Comment:
Is it reasonable to wrap this in a `try`/`catch` just in case we meet a library that dislikes this signature?
```suggestion
          let providedLogger: Logger = client["logger"];

          try {
            // create a child logger if the provided logger has child logger implementation
            if ("child" in providedLogger) {
              providedLogger = (
                providedLogger.child as (
                  metadata: Record<string, unknown>
                ) => Logger
              )(metadata);
            }
          } catch (err) {
            // no-op
          }

```

Reply:
sure, we can add it, but would like it to be something more readable instead of mixing both type declaration with func execution.